forum/qa-content/javascript/imgpre.js

```diff
@@ -21,12 +22,12 @@
 	];
 
 	window.addEventListener('load', function() {
-		
+

```

Comment:
Ogólnie, to ten cały skrypt jest IMO dziwny. Doczytuje obrazki, które wyglądają na używane do stanu `:hover` na obrazkach w sidebarze kategorii, które z kolei są wcześniej ładowane bezpośrednio w HTML-u :thinking:

Reply:
Prawdę mówiąc nie za bardzo zastanawiałem się co on robi, ja tylko zaktualizowałem kategorie. Może jest niepotrzebny? Z resztą pamiętam, że on już kiedyś był poprawiany, bo początkowo w ogóle wyglądał słabo.

Reply:
Bez tego skryptu nie załadują się dodatkowe obrazki. Tylko nie wiem dlaczego akurat te obrazki mają być doczytywane asynchronicznie - zakładam, że dlatego, że pokazują się na `:hover`, czyli właściwie jest mała szansa, że user zechce je zobaczyć w trakcie ładowania strony, albo w ogóle. Więc mogą się załadować leniwie.

IMO można to poprawić, tj. zainicjować pobieranie, gdy obszar tych obrazków znajdzie się w viewporcie, a na mobilce ich nie pobierać, bo z tego co widzę, one w ogóle nie są wtedy widoczne.

Reply:
> IMO można to poprawić, tj. zainicjować pobieranie, gdy obszar tych obrazków znajdzie się w viewporcie, a na mobilce ich nie pobierać, bo z tego co widzę, one w ogóle nie są wtedy widoczne.

Zapewne zależało autorom na tym, aby pobierać dany obrazek gdy jest on potrzebny. Nie muszą wtedy przewidywać czy user chce na to najechać czy nie, tylko w zależności od neta żądać obrazka. 

To chyba Jedno optymalizacyjnych rozwiązań, aby nie wyczerpać zasobów serwera + poprawić szybkość strony.

Poprawiony skrypt możemy uznać, że działa gdy @awaluk  nie wpadnie za dwa tygodnie, że skoczyły statystyki pobieranych zasobów ;D

Reply:
> tylko w zależności od neta żądać obrazka.

Nie rozumiem gdzie tutaj jest jakaś logika zależąca od połączenia sieciowego (bo masz na myśli internet mobilny vs _zwykły_)?

> To chyba Jedno optymalizacyjnych rozwiązań, aby nie wyczerpać zasobów serwera + poprawić szybkość strony.

Co do niewyczerpywania serwerów to nie wiem, bo zależy czy ładowanie obrazków w ten sposób jest cache'owane przez przeglądarkę. Jeśli nie jest, to z perspektywy serwera raczej nie różni się niczym od pobrania obrazka w zwykły sposób. Co do poprawy szybkości ładowania strony, to już prędzej, bo obrazki przez ten skrypt są ładowane asynchronicznie po załadowaniu całej strony, łącznie z zasobami (event [`load`](https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event)).

> Zapewne zależało autorom na tym, aby pobierać dany obrazek gdy jest on potrzebny.

Nie znam intencji autorów - z treści PRa nie wiele wynika. Ale nie bardzo zgodzę się z tym, że obrazki pobierane są wtedy, gdy są potrzebne, bo ten skrypt pobiera je bezwarunkowo, gdy strona się załaduje - niezależnie czy użytkownik będzie chciał obejrzeć obrazki przewijając stronę, czy będzie najeżdżał na nie myszką, czy od razu przejdzie do innej strony (np. klikając w pierwszy temat albo któryś z linków nawigacyjnych na górze strony).

Reply:
Nie myślałeś o tym aby zostawić jednak przy C# tego .neta?
Może warto dodać do JavaScript również TypeScript?